pypeit/images/processrawimage.py

```diff
@@ -219,7 +219,9 @@ def process(self, process_steps, pixel_flat=None, illum_flat=None, bias=None):
             steps_copy.remove('apply_gain')
         # Flat field
         if 'flatten' in process_steps:
```

Comment:
The control flow probably needs to be revisited.  I think the reason this is like this at the moment is that we effectively want to allow for `'flatten'` to be included in the processing steps (i.e., you should always flat field the data), but allow for the code to continue if the pixel flat hasn't been created because of some fault.  We could instead test that the pixel flat is successful before adding `flatten` to the list of processing steps, or have pixel-flat set to ones if it's provided as None.  I think we need to avoid allocating arrays as much as possible, which is why it's this way at the moment.  But we can revisit after this PR.